var/spack/repos/builtin/packages/ccs-qcd/package.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class CcsQcd(MakefilePackage):
+    """This program benchmarks the performance of a linear equation solver
+       with a large sparse coefficient matrix appering in a lattice QCD
+       probrem. Lattice QCD describes the property of protons, neutrons and
+       neucleons in terms of more fundamental interacting elementary particles
+       gluon and quarks."""
+
+    homepage = "https://github.com/fiber-miniapp/ccs-qcd"
+    git      = "https://github.com/fiber-miniapp/ccs-qcd.git"
+
+    version('master', branch='master')
+    version('1.2.1', commit='d7c6b6923f35a824e997ba8db5bd12dc20dda45c')
+
+    variant(
+        'class',
+        default='1',
+        values=(
+            '1',  # 8x8x8x32 (default MPI config: 1x1x1)
+            '2',  # 32x32x32x32 (default MPI config: 4x4x4)
+            '3',  # 64x64x64x32 (default MPI config: 8x8x8)
+            '4',  # 160x160x160x160 (default MPI config: 20x20x20)
+            '5',  # 256x256x256x256 (default MPI config: 32x32x32)
+            '6'   # 192x192x192x192 (default MPI config: 24x24x24)
```

Comment:
You can make these ints instead of strings.

Reply:
I think I've learned a lot.
I made these ints instead of strings and added error messages which are occurred when we choose `the value of class < 1` or `the value of class > 6`.